src/rdf4cpp/rdf/datatypes/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.17)
+
+add_library(rdf4cpp_datatypes OBJECT
+        owl/Rational.cpp
+        owl/Real.cpp
+        rdf/LangString.cpp
+        xsd/integers/non_positive/NegativeInteger.cpp
+        xsd/integers/non_positive/NonPositiveInteger.cpp
+        xsd/integers/non_negative/NonNegativeInteger.cpp
+        xsd/integers/non_negative/PositiveInteger.cpp
+        xsd/integers/non_negative/UnsignedByte.cpp
+        xsd/integers/non_negative/UnsignedInt.cpp
+        xsd/integers/non_negative/UnsignedLong.cpp
+        xsd/integers/non_negative/UnsignedShort.cpp
+        xsd/integers/non_positive/NegativeInteger.cpp
+        xsd/integers/non_positive/NonPositiveInteger.cpp
+        xsd/integers/signed/Byte.cpp
+        xsd/integers/signed/Int.cpp
+        xsd/integers/signed/Integer.cpp
+        xsd/integers/signed/Long.cpp
+        xsd/integers/signed/Short.cpp
+        xsd/Boolean.cpp
+        xsd/Decimal.cpp
+        xsd/Double.cpp
+        xsd/Float.cpp
+        xsd/String.cpp
+        )
+
+target_include_directories(
+        rdf4cpp_datatypes
+        PRIVATE
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>/../../.."
```

Comment:
If I understand the rdf4cpp_datatypes library right, it exposes no headers and just compiles the datatype classes. 
You could even make the target_link_libraries private then. 

What is the rational behind this separate cmake file? Does the private include speed up the build time? If not the effective make files resulting from cmake should be pretty much the same as if the datatype cpp would be directly part of rdf4cpp